python/cudf/dataframe/series.py

```diff
@@ -1479,6 +1481,28 @@ def digitize(self, bins, right=False):
 
         return Series(numerical.digitize(self._column, bins, right))
 
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
+        """Shift values of an input array by periods positions and store the
+        output in a new array.
+
+        Notes
+        -----
+        Shift currently only supports float and integer dtypes.
+        """
+        assert axis in (None, 0) and freq is None and fill_value is None
+
+        if not np.issubdtype(self.dtype, np.number):
+            raise NotImplementedError("Shift currently only supports "
+                                      "numeric dtypes")
+        if periods == 0:
+            return self
+
+        input_dary = self.data.to_gpu_array()
+        output_dary = rmm.device_array_like(input_dary)
+        cudautils.gpu_shift.forall(output_dary.size)(input_dary, output_dary,
+                                                     periods)
+        return Series(output_dary, name=self.name, index=self.index)
```

Comment:
This won't handle nulls in its current state. Suggest either adding a check for `null_count == 0` or implementing a bitshift to handle the nulls.